### ISSUE TITLE
BBC : Struct::Dumb

### ISSUE THREAD

**Issue by shlomif** [Perl/perl5#20801]
Cc: shlomif@telaviv1.shlomifish.org
To: perl5-porters@perl.org
From: shlomif@gmail.com
Reply-To: shlomif@gmail.com
Subject: BBC: Struct::Dumb
Message-Id: <5.37.9_365695_1676108555@telaviv1.shlomifish.org>

This is a bug report for perl from shlomif@gmail.com,
generated with the help of perlbug 1.43 running under perl 5.37.9.

-----------------------------------------------------------------

**Description**

**Steps to Reproduce**

**Expected behavior**

---
**Flags**
- category=core
- severity=low
---
**Perl configuration**
```
Site configuration information for perl 5.37.9:

Configured by shlomif at Sat Feb 11 07:16:21 IST 2023.

Summary of my perl5 (revision 5 version 37 subversion 9) configuration:
  Local Commit: c7342b06c24013ac518d5817617c3abbef7f0a1e
  Ancestor: fc931ecfefd2e26fc39f6d7a50d7abf611c19f52
  Platform:
    osname=linux
    osvers=6.1.10-200.fc37.x86_64
    archname=x86_64-linux-thread-multi
    uname='linux telaviv1.shlomifish.org 6.1.10-200.fc37.x86_64 #1 smp preempt_dynamic mon feb 6 23:56:48 utc 2023 x86_64 x86_64 x86_64 gnulinux '
    config_args='-de -Dprefix=/home/shlomif/apps/perl/bleadperl -Dusedevel -Dusethreads'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='cc'
    ccflags ='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2'
    optimize='-O2'
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion=''
    gccversion='12.2.1 20221121 (Red Hat 12.2.1-4)'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib /usr/lib64 /usr/local/lib64
    libs=-lpthread -ldb -ldl -lm -lcrypt -lutil -lc
    perllibs=-lpthread -ldl -lm -lcrypt -lutil -lc
    libc=/lib/../lib64/libc.so.6
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version='2.36'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E'
    cccdlflags='-fPIC'
    lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector-strong'

Locally applied patches:
    c0a24b11a0434ad8c8b1e758b952f8546d69c216
    85d2c8e80f418c456c28213a08f692e33833b8c4
    42a4d650822086a91f0da204e86e1bc35c2ed66e
    e9f20e0860ca7ba8a9430281f0c0e3dc4ba912cf
    d404f73baa29d5697ed2cac768de015217d5e819
    6abce8468c57a533bc8160fc30734708c2452aea
    5dd33ebbb550440bb773b780efe02646f3c6b10a
    02b54f9d07226e99a30271314e02617d5629b511
    93688065283a7ee7f0914a6448f7ea9209342f00
    2d51effbcbd1f6edbafa5974996a9d244353dd25
    85fcefb858d7287b1c90277ead7d1cdd170e48d0
    a36fec492e3c37aae28f47766892f34b74d51b31
    7e2d91e6d3a09e2ebb61242bb18ff95d30d9560d
    dcf36a4fb9625391d047f5883d41b608a40f3382
    83ba55794efd51e7a10531c7e50a859fe310ffb0
    aa2d00d7d06d0ec6110c4351b635728dfa53bc81
    70bac31a4f4e90f39d9876a34715f0c83d2684dd
    3e1c78bda6bcaff0d1d04c51c98ff3ea0ba0db8b
    ce8f481253f8f6250878c58247f4d8fee6a3a3ce
    92373dea9d7bcc0a017f20cb37192c1d8400767f
    923d5d2113e738eea21d56eb8f1589c6b3506f02
    dead3ccedb0e57c37846b3b30aa1d618ef1309f3
    c48bc75b4b72b05c98d37de9e34db008b02b0819
    61c36998dd2ab81a3a53358ac76a57e6f5850dc9
    9312140ba1ac523a74c69d552697f2767e128e04
    66c1d2a57eb49d2facb14a07e915cf285e732f21
    9baf61eb9f3672d1f743ea0f4e5014045e15ea02
    376c059310e74ce8b9f3f45a3e9f92160939ca60
    84e73d4ea20306de42377d1363057e9baa0a10a1
    0de69e372d3baf843522c30d8711b6513c2a472b
    626df4fd56759528a170e821cab27f372b174496
    6c19911f93fc29afe94cad617c6f2f3325ad77ba
    b1414dcb98d72036e17cb9d4c8aae9d7ad65113e
    dd3d299fefbbff9efe55eb16af5bb9aa0410d606
    9d3a26c75c777cb4dc1382d59de52a9b69a38f14
    71d63d0dc1fcf23d28f488655c105c0dfefbd254
    4202141d20ddfa0501f385cf923860bcf7511398
    5f5c35d3ce139755d02fa07de26229cd08d3c8cd
    c05434dd5e129bbae33020bc1d041ecb058215ae
    2e75d796b38ec046115bbbb9a22c7a966467ade5
    841a3e1bca89544d115c2d2d3df1c96d27172a15
    ac4290d01742ee1228041e7467ed23414ebfbafc
    a435da2b160255de4884b0d007ef0d9d93ec203b
    457ba501e6d4253da0c5acc6c45c96c598b3f605
    73bdb838da96b52da71c9080ff6fdc2e2c88ea23
    c3be9bd4d181af38c0c9e2ae3c5ee51d49651eff
    d7c2c45e0afe114a6fc55207b26d268ec4affea6
    bba8d7cfa7a094d55326d7a9cfebd73fb8ab2a35
    03fc59afb4069bd5f569c5c4fb2e7b4293d948a9
    0880e953fabd23c1207fab60ed2647e122a4e99a
    ea18e4f378a48b29f06181f2c32101dfc7205ba0
    5e4b76a1263a3865a4d21af871474e3c6e790362
    a8a9fd0a672789f304a0ecb83240b4e6d34f1d1f
    93425aaa4c6367b09eef90a3a7bb2e7c9ab072ff
    7dccff44c3ee4d9e1d201c6892ec9c38746975b7
    02bb7cad54cd703b780436e2129bdb299457105b
    bd90fa412bb2eb27ffa23a0e439350a39847521e
    efa8f708cd0da7c71bb71920a47b545bd07bc28a
    107e6fb3d831b51ef9e6cca524d17a8c1717d0ab
    fdce04eede5d38fa6df55c4461e7b833c9d68aba
    fe7859256d3a8e646ec90c7313a105c8d273db32
    472953645a796ce6fd09edf90e294f612584e71d
    211b33a3986847f6d6cec8e83ba355f8040fa039
    e7252fd45274f6b0c6e32ef76b6e0e325eb5ce5b
    35b3adc04efc1e4fd37c5101758174eb0615a409
    77b6579f19b9a3ca5c29e5b37dfc66c132882745
    a98bbd4e296484149b8a25a893a1529dc5d64c4a
    b162e59c910a96e062c8c37d6270fe38b2db3383
    8ab54e77a59c2f25aff25440e4592404cc8d8d72
    5d4d8b9fa5103e3bd911bb1ab9e94eb6d8e0ff28
    99b497aa90ed7db99d29a301b47c91fba65c9cb3
    24c33697796a1556af3f58e15fc4fb6b0d1538dc
    ca8d92cfddc9cc3d4904712f7e0dfc3b2598561d
    fd11211b37c20262922903f6435c1c09ac046e5c
    b40895ae558e0aff0c347785dafeaaff40a01801
    69953ef3f17b02d83fb922e406db19fbd39f6fa3
    9bf25cf083493be6ba4df71fd884626369153fce
    e51627afd15d704290c8201fdfc02bd7951564f3
    0bb17957e5474866af5b126cf584f2f7aa6340fb
    25d7fad339e5b78ac9a2a40f147a903c1aedf095
    e3b003dd8ca8d881450f6c96297ffa35f6a92ed1
    054ceeebc7f6acad2772faa4fa88617c35f4a88a
    311ca5baa9210d88e6008e4989d0907e2b5e4982
    07c4c053ee9ee2dac36d7eadd6d2558d94a7a802
    d8b29a3430b219e3ab3dae2947a0ff22885c1b5e
    b45c1f8aeb97b765fe1c30fe69a0b0bbff9d9925
    f2244cf0c11cc95672ad906ead231fe5a20e965e
    1863a6c87cb85549953c181e1d6a060138d60337
    2ba5f4dfa884b4b8257dcbb4213ed932e82113a2
    a170805d60a9631c88faf0da744391bfc8f2fe26
    a683fa5b7e816ae5c10d246c9a0b1f3ea743274b
    94a21c89d725dee58126a57a36a33a72e3dd54fd
    8d83c9f712bbdad0e46a7ea27be5959e9a3e4116
    28714c877af89d6ee91eedcb5a8bee185b20360a
    e27772f9bca7582d502812a94e3638f7f2a269b1
    90483ee5ba0858ac3f0e5fd6e39ccc3202001749
    2aa450394fed0a447d8df334bc797a50a81aebe3
    8da6832dfbca54f8380d08a45b2ca920607bf04c
    34ac82665ce776460beed5d48a4e1acc4943588c
    2664c462411478c2f23e27a664b4b92f10b7453f
    c7342b06c24013ac518d5817617c3abbef7f0a1e

---
@INC for perl 5.37.9:
    /home/shlomif/apps/perl/bleadperl/lib/site_perl/5.37.9//x86_64-linux-thread-multi
    /home/shlomif/apps/perl/bleadperl/lib/site_perl/5.37.9/
    /home/shlomif/apps/perl/bleadperl/lib/5.37.9//x86_64-linux-thread-multi
    /home/shlomif/apps/perl/bleadperl/lib/5.37.9/
    /home/shlomif/apps/perl/bleadperl/lib/site_perl/5.37.9/x86_64-linux-thread-multi
    /home/shlomif/apps/perl/bleadperl/lib/site_perl/5.37.9
    /home/shlomif/apps/perl/bleadperl/lib/5.37.9/x86_64-linux-thread-multi
    /home/shlomif/apps/perl/bleadperl/lib/5.37.9

---
Environment for perl 5.37.9:
    HOME=/home/shlomif
    LANG=en_GB.UTF-8
    LANGUAGE=en_GB:en
    LC_ADDRESS=en_US.UTF-8
    LC_COLLATE=en_US.UTF-8
    LC_CTYPE=en_US.UTF-8
    LC_IDENTIFICATION=en_GB.UTF-8
    LC_MEASUREMENT=en_GB.UTF-8
    LC_MESSAGES=en_US.UTF-8
    LC_MONETARY=en_US.UTF-8
    LC_NAME=en_GB.UTF-8
    LC_NUMERIC=en_GB.UTF-8
    LC_PAPER=en_US.UTF-8
    LC_TELEPHONE=en_US.UTF-8
    LC_TIME=en_GB.UTF-8
    LD_LIBRARY_PATH=:/home/shlomif/apps/to-del-ncurses/lib
    LOGDIR (unset)
    PATH=/home/shlomif/Download/unpack/perl/p5/emcc-build/perl-emcc-build:/home/shlomif/apps/perl/bleadperl/bin:/home/shlomif/bin:/usr/local/bin:/usr/bin:/usr/local/games:/usr/games:/usr/lib64/qt4/bin:/usr/lib64/qt5/bin:/bin
    PERL5LIB=:/home/shlomif/apps/perl/bleadperl/lib/site_perl/5.37.9/:/home/shlomif/apps/perl/bleadperl/lib/5.37.9/
    PERLBREW_ROOT=/home/shlomif/apps/perl/brew
    PERL_BADLANG (unset)
    PERL_MM_USE_DEFAULT=1
    SHELL=/bin/zsh
```

**Comment by demerphq** [comment]
FWIW, it works at v5.37.8-91-g5d4d8b9fa5 aka 5d4d8b9fa5103e3bd911bb1ab9e94eb6d8e0ff28

**Comment by leonerd** [comment]
Yeah; upstream module bug. Fixed in Struct::Dumb 0.14.
